File: server/internal/model/hit_analysis.go
// server/internal/model/hit_analysis.go
package model

import "time"

// HitAnalysis 爆款拆解记录
type HitAnalysis struct {
	ID          uint      `gorm:"primaryKey" json:"id"`
	UserID      uint      `gorm:"index" json:"user_id"`
	PortfolioID uint      `gorm:"index" json:"portfolio_id"`
	Title       string    `gorm:"size:200;not null" json:"title"`        // 被拆解的小说标题
	Author      string    `gorm:"size:100" json:"author"`
	SourceText  string    `gorm:"type:longtext" json:"source_text"`      // 分析素材（完成后清除）
	Report      string    `gorm:"type:longtext" json:"report"`           // JSON: 结构化拆解报告
	WorkflowID  uint      `gorm:"index;default:0" json:"workflow_id"`
	Status      string    `gorm:"size:20;default:pending" json:"status"` // pending/running/completed/failed
	ModelName   string    `gorm:"size:50" json:"model_name"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
}

File: web/src/api/hit_analysis.ts
// web/src/api/hit_analysis.ts
import request from './request'

// ========== 类型定义 ==========

export interface HitAnalysis {
  id: number
  user_id: number
  portfolio_id: number
  title: string
  author: string
  source_text: string
  report: string // JSON string
  workflow_id: number
  status: string // pending/running/completed/failed
  model_name: string
  created_at: string
  updated_at: string
}

export interface HitAnalysisReport {
  structure_analysis: string
  rhythm_analysis: string
  character_arcs: string
  hook_points: { position: string; type: string; technique: string }[]
  style_features: string
  summary: string
}

// ========== API ==========

export const hitAnalysisApi = {
  submit: (data: { portfolio_id: number; title: string; author?: string; source_text: string; model_name?: string }) =>
    request.post('/hit-analysis', data),
  list: () => request.get('/hit-analysis'),
  get: (id: number) => request.get(`/hit-analysis/${id}`),
  delete: (id: number) => request.delete(`/hit-analysis/${id}`),
}
